FileConv/Gfx/PrintShopFont.cs

```diff
@@ -0,0 +1,416 @@
+                    }
+                }
+            }
+            return Applicability.Not;
+        }
+
+        public override Type GetExpectedType(Dictionary<string, string> options)
+        {
+            return typeof(IBitmap);
+        }
+
+        public override IConvOutput ConvertFile(Dictionary<string, string> options)
+        {
+            if (Applic <= Applicability.Not)
+            {
+                Debug.Assert(false);
+                return new ErrorText("ERROR");
+            }
+
+            Debug.Assert(DataBuf != null);
+            Debug.Assert(DataBuf.Length >= 0);
+            Debug.Assert(DataStream != null);
+
+
+            bool doMult = GetBoolOption(options, OPT_MULT, false);
+
+            return ConvertMono(DataBuf, Palette8.Palette_MonoBW, doMult);
+        }
+
+        /// <summary>
+        /// Converts a Print Shop font file to a B&amp;W bitmap.
+        /// </summary>
+        public Bitmap8 ConvertMono(byte[] buf, Palette8 palette, bool doMult)
+        {
+            Bitmap8 output;
+
+            int totalWidth = 0;
+            int maxHeight = 0;
+
+            for (int idx = 1; idx < CHARACTER_COUNT; idx++)
+            {
+                if (idx != 32)
+                {
+                    if (idx != 1) totalWidth += INTER_CHARACTER_SPACING_PX;
+                    totalWidth += GetWidthForCharacter(idx);
+                    int height = GetHeightForCharacter(idx);
+                    if (height > maxHeight) { maxHeight = height; }
+                }
+            }
+
+            Debug.WriteLine("TotalWidth: " + totalWidth);
+
+            if (doMult)
+            {
+                output = new Bitmap8((4 + totalWidth) * XMULT, (4 + maxHeight) * YMULT);
+                Debug.WriteLine("Bitmap3: Width: " + (4 + totalWidth) * XMULT + " Height: " + (4 + maxHeight) * YMULT);
+            }
+            else
+            {
+                output = new Bitmap8(totalWidth + 4, maxHeight + 4);
+                Debug.WriteLine("Bitmap: Width: " + totalWidth + 4 + " Height: " + maxHeight + 4);
+            }
+            output.SetPalette(palette);
+
+            int xOffset = 1;
+            for (int idx = 1; idx < CHARACTER_COUNT; idx++)
+            {
+                if (idx != 32)
+                {
+                    byte charWidth = GetWidthForCharacter(idx);
+                    byte charHeight = GetHeightForCharacter(idx);
+                    byte[] charData = GetDataForCharacter(idx);
+
+                    int offset = 0;
+                    for (int row = 0; row < charHeight; row++)
+                    {
+                        for (int col = 0; col < WidthToBytes(charWidth); col++)
+                        {
+
+                            byte bval = charData[offset++];
+                            for (int bit = 0; bit < 8; bit++)
+                            {
+                                // Bits are zero for white, one for black; leftmost pixel in MSB.
```

Comment:
It looks like fonts use 0 for the background, 1 for the foreground. Should this definition be reversed?  Print Shop would treat the background as white, since it's printing on paper, but fonts are generally shown as white on a black background when viewed on a computer.  (I don't have an opinion here, just asking the question aloud.)

For a future update, you may want to use an additional color (possibly transparent) for the gaps between characters.  Further, you could render them in a 10x6 grid (similar to the way hi-res fonts are done), using transparent pixels to fill the variable-width spaces.  That would make the font easier to see in the file viewer.  Certainly not necessary; seeing the first few characters is enough to give a sense for the font style.

Reply:
I've changed it so that it is black on white, mirroring the clip art code.  
In a future update, I plan on implementing an option to toggle between a linear or gridded display.